### PR TITLE
Fix TUP-21510 Update PasswordEncryptUtil

### PR DESCRIPTION
**Pull request by kjwang-talend**
Fix TUP-21510 Update PasswordEncryptUtil
https://jira.talendforge.org/browse/TUP-21510

### CODE REVIEW

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java [hunk 1]
```diff
@@ -160,4 +161,19 @@ public static String getPasswordDisplay(String value) {
         }
     }
 
+    public static String aesEncryptPassword(String input, String key) throws Exception {
```
Comment (coheigea) [inline main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: Where are we planning to use these methods? Will there be a separate PR to update them?

File: main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        Encryption encryption = getEncryption(key);
+        return encryption.decrypt(input);
+    }
+
+    public static String decryptPassword(String input) throws Exception {
+        Encryption encryption = getEncryption();
+        return encryption.decrypt(input);
+    }
+
+    public static Encryption getEncryption() {
```
Comment (coheigea) [inline main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java]: Can the getEncryption() methods be made private? They're not used outside this class.

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java (hunk 1 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: I have removed it. thanks

File: main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java (hunk 2 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java]: Yes, improved. Thanks

File: main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public static String decryptPassword(String input) throws Exception {
+        Encryption encryption = getEncryption();
+        return encryption.decrypt(input);
+    }
+
+    private static Encryption getEncryption() {
+        if (defaultEncryption == null) {
+            defaultEncryption = new Encryption(KeySources.fixedKey(ENCRYPTION_KEY), CipherSources.aes());
```
Comment (coheigea) [inline main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java]: Better to write: defaultEncryption = getEncryption(ENCRYPTION_KEY);

File: main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+import org.talend.daikon.crypto.CipherSources;
+import org.talend.daikon.crypto.Encryption;
+import org.talend.daikon.crypto.KeySources;
+
+public class AESEncryption {
```
Comment (coheigea) [inline main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java]: Add a comment explaining what the class does. Maybe add a TODO to remove the hard-coded password for the next task.

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java [hunk 5]
```diff
@@ -108,12 +109,7 @@ public static String encryptPasswordHex(String input) throws Exception {
         if (input == null) {
             return input;
         }
-        SecretKey key = getSecretKeyUTF8();
```
Comment (coheigea) [inline main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: Can getSecretKeyUTF8() be removed as well?

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java [hunk 6]
```diff
@@ -108,12 +109,7 @@ public static String encryptPasswordHex(String input) throws Exception {
         if (input == null) {
```
Comment (coheigea) [inline main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: I think the method should be renamed, as the output is no longer HEX, but BASE-64. Better to call it something generic instead. Or will this change too many other source files?

File: main/plugins/org.talend.librariesmanager/resources/java/routines/system/PasswordEncryptUtil.java [hunk 7]
```diff
@@ -26,55 +23,36 @@
 
     public static String ENCRYPT_KEY = "Encrypt"; //$NON-NLS-1$
 
-    private static String rawKey = "Talend-Key"; //$NON-NLS-1$
-
-    private static SecretKey key = null;
-
-    private static SecureRandom secureRandom = new SecureRandom();
+    private static final String ENCRYPTION_KEY = "Encrypt-Talend-TalendKey";
```
Comment (coheigea) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/system/PasswordEncryptUtil.java]: Do we need to define ENCRYPTION_KEY again, seeing as it's defined in AESEncryption? It should only be defined in one place.

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java (hunk 5 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: ok, thanks for your suggestion

File: main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java (hunk 4 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java]: ok. thanks

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java (hunk 6 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: It related serval repositories. Can we just keep it now?

File: main/plugins/org.talend.librariesmanager/resources/java/routines/system/PasswordEncryptUtil.java (hunk 7 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.librariesmanager/resources/java/routines/system/PasswordEncryptUtil.java]: Yes, we need to define it again. Because this class will be built out with user's job archive. We'd better don't depend on any libraries I think.

File: main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java (hunk 3 above)
Comment (kjwang-talend) [reply main/plugins/org.talend.utils/src/main/java/org/talend/utils/security/AESEncryption.java]: Done, Thanks for your suggestion.

File: main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java (hunk 6 above)
Comment (coheigea) [reply main/plugins/org.talend.commons.runtime/src/org/talend/commons/utils/PasswordEncryptUtil.java]: Yep fine.

**Comment by kjwang-talend** [comment]
> 
> 
> Approved, but please create a JIRA when done to change the hard-coded password for 7.3.1.

The JIRA ticket : https://jira.talendforge.org/browse/TUP-22966. Thanks
